### ISSUE TITLE
False positive for tslint:explicit-override rule after upgrading to TypeScript 4.3

### ISSUE THREAD

**Issue by dominic-simplan** [typescript-eslint/typescript-eslint#3646]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

.eslintrc.js
```JSON
{
  "rules": {
    "@typescript-eslint/tslint/config": [
            "error",
            {
                lintFile: "./tslint.json",
            },
        ],
  }
}
```

tslint.json:
```JSON
{
    "extends": ["tslint-override"],
    "defaultSeverity": "off",
    "rules": {
        "explicit-override": {
            "severity": "error"
        }
    }
}
```

```TS
export interface Base {
  baseMethod?(): void;
}

export class MyClass implements Base {
  /** @override */ public baseMethod(): void {
  }
}
```

**Expected Result**

Should not show an ESLint warning/error as override comment is added correctly.

**Actual Result**
With TypeScript < 4.3, this works correctly and does not show an ESLint warning/error.
With TypeScript = 4.,3.x, this shows the warning/error:

> Member is overriding a base member. Use the @override keyword if this override is intended (tslint:explicit-override)eslint@typescript-eslint/tslint/config)

**Additional Info**

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin-tslint` | `4.28.3` |
| `@typescript-eslint/parser`               | `4.28.3` |
| `TypeScript`                              | `4.3.5` |
| `ESLint`                                  | `7.31.0` |
| `node`                                    | `14.16.0` |
| `tslint` | `6.1.3` |`
| `tslint-override`  | `1.0.1` |

**Comment by bradzacher** [comment]
A few things.

1) TSLint is not part of this project, we just provide the tooling to integrate it into ESLint.
2) TSLint is also deprecated. It is no longer maintained. https://github.com/palantir/tslint
   You should endeavour to migrate all your usages to ESLint and remove TSLint from your project.
3) the rule you're talking about isn't even part of core TSLint - it's a custom 3rd-party rule, as evidenced by your listing of the package in the versions table. https://github.com/hmil/tslint-override

**Comment by dominic-simplan** [comment]
Yes, sorry for the ticket. I was under the impression that the cause could be within this repository, as I saw that changes had been made to adjust the library for TS 4.3. Probably it is the tslint-override rule which doesn't work anymore with the newest TS.

I agree with your advice, however it is not so easy to migrate as discussed in #3608... :-)

Thanks for taking the time to explain and comment!
